agent/containermetadata/parse_metadata.go

```diff
@@ -40,17 +42,31 @@ func (manager *metadataManager) parseMetadataAtContainerCreate(taskARN string, c
 // configuration and data then packages it for JSON Marshaling
 // Since we accept incomplete metadata fields, we should not return
 // errors here and handle them at this or the above stage.
-func (manager *metadataManager) parseMetadata(dockerContainer *docker.Container, taskARN string, containerName string) Metadata {
-	dockerMD := parseDockerContainerMetadata(taskARN, containerName, dockerContainer)
+func (manager *metadataManager) parseMetadata(dockerContainer *docker.Container, task *api.Task, containerName string) Metadata {
+	taskDefinitionRevision := parseTaskDefinitionRevision(task, containerName)
+	dockerMD := parseDockerContainerMetadata(task.Arn, containerName, dockerContainer)
 	return Metadata{
 		cluster:                 manager.cluster,
-		taskMetadata:            TaskMetadata{containerName: containerName, taskARN: taskARN},
+		taskMetadata:            TaskMetadata{containerName: containerName, taskARN: task.Arn, taskDefinitionFamily: task.Family, taskDefinitionRevision: taskDefinitionRevision},
 		dockerContainerMetadata: dockerMD,
 		containerInstanceARN:    manager.containerInstanceARN,
 		metadataStatus:          MetadataReady,
 	}
 }
 
+// parseTaskDefinitionRevision converts the task.Version from a string to an int.
+// this allows us to match the type used in our CLI:
+// https://docs.aws.amazon.com/cli/latest/reference/ecs/describe-task-definition.html
+// if anything bad happens, then 0 is returned, so you must omitempty to omit this from
+// the metadata JSON output.
+func parseTaskDefinitionRevision(task *api.Task, containerName string) int {
```

Comment:
as per my comment in `containermetadata/types.go`, i think this has to return (*int, error). If there's an error, the pointer would not be set and omitempty will kick in.